ui/permissions_test.py

```diff
@@ -32,28 +59,95 @@ def collection():
+def request_data():
+    """
+    Fixture for tests requiring a user, request, and view
+    """
+    user = factories.UserFactory()
     request = Mock(
+        method='GET',
+        user=user
+    )
+    view = Mock(kwargs={'user': user.username})
+    return SimpleNamespace(user=user, request=request, view=view)
+
+
+@pytest.fixture
+def request_data_su():
+    """
+    Fixture for tests requiring a superuser, request, and view
+    """
+    user = factories.UserFactory()
+    user.is_superuser = True
+    user.save()
+    request = Mock(
+        method='GET',
+        user=user
+    )
+    view = Mock(kwargs={'user': user.username})
+    return SimpleNamespace(user=user, request=request, view=view)
+
+
+@pytest.fixture
+def request_data_anon():
+    """
+    Fixture for tests requiring an anonymous user, request, and view
+    """
+    request = Mock(
+        method='GET',
         user=AnonymousUser()
     )
     view = Mock(kwargs={'user': 'username'})
-    assert collection_owner_permission.has_object_permission(request, view, collection) is False
+    return SimpleNamespace(request=request, view=view)
+
+
+@pytest.fixture
+def alt_moira_data():
+    """
+    Fixture for tests requiring a moira list, collection, and user
+    """
+    alt_moira_list = MoiraListFactory()
+    alt_collection = CollectionFactory()
+    alt_user = factories.UserFactory()
+    return SimpleNamespace(moira_list=alt_moira_list, collection=alt_collection, user=alt_user)
+
+
+def test_is_collection_owner_anonymous_users(collection_owner_permission, collection, request_data_anon):
+    """
+    Test for IsCollectionOwner.has_object_permission to verify that anonymous users do not have permission
+    """
+    assert collection_owner_permission.has_object_permission(request_data_anon.request,
+                                                             request_data_anon.view,
+                                                             collection) is False
 
 
-def test_is_collection_owner_user_different_from_collecton_owner(collection_owner_permission, collection):
+def test_is_collection_owner_user_different_from_collecton_owner(collection_owner_permission,
+                                                                 collection,
+                                                                 request_data):
     """
     Test for IsCollectionOwner.has_object_permission to verify
     that user who does not own the collection does not have permission
     """
-    other_user = factories.UserFactory()
-    request = Mock(
-        user=other_user
-    )
-    view = Mock(kwargs={'user': other_user.username})
-    assert collection_owner_permission.has_object_permission(request, view, collection) is False
+    assert collection_owner_permission.has_object_permission(request_data.request,
+                                                             request_data.view,
+                                                             collection) is False
```

Comment:
In this case it seems the fixture just made the test more obtuse, how about just adding this additional assert just above it to make it clear the users are different:
`assert request_data.user != collection.owner`